DuckDuckGo/FireproofFaviconUpdater.swift

```diff
@@ -57,11 +57,11 @@ class FireproofFaviconUpdater: NSObject, FaviconUserScriptDelegate {
     let tab: TabNotifying
     let favicons: FaviconProviding
 
-    init(bookmarksDatabase: CoreDataDatabase, tab: TabNotifying, favicons: FaviconProviding) {
+    init(bookmarksDatabase: CoreDataDatabase, tab: TabNotifying, favicons: FaviconProviding, secureVaultEnabled: Bool) {
         self.context = bookmarksDatabase.makeContext(concurrencyType: .mainQueueConcurrencyType)
         self.tab = tab
         self.favicons = favicons
-        secureVault = try? SecureVaultFactory.default.makeVault(errorReporter: SecureVaultErrorReporter.shared)
+        self.secureVault = secureVaultEnabled ? try? SecureVaultFactory.default.makeVault(errorReporter: SecureVaultErrorReporter.shared) : nil
```

Comment:
I think we should be fine for now with ignoring the error being reported and just ignore the lack of secure vault.
At the same time, this means we may want to try and create secureVault if it's nil at the moment we execute the Favicon removal logic (also checking the feature flag and user setting in that place, as I think it's fine there).